runtime/compiler/control/CompilationThread.cpp

```diff
@@ -10466,71 +10483,116 @@ TR::CompilationInfo::increaseUnstoredBytes(U_32 aotBytes, U_32 jitBytes)
 #endif
    }
 
+
 uint64_t
 TR::CompilationInfo::computeFreePhysicalMemory(bool &incompleteInfo)
    {
    bool incomplete = false;
+   PORT_ACCESS_FROM_JITCONFIG(_jitConfig);
 
-   if (_cachedFreePhysicalMemoryB >= 0)
+   uint64_t freePhysicalMemory = OMRPORT_MEMINFO_NOT_AVAILABLE;
+   J9MemoryInfo memInfo;
+   if (0 == j9sysinfo_get_memory_info(&memInfo)
+      && memInfo.availPhysical != OMRPORT_MEMINFO_NOT_AVAILABLE
+      && memInfo.hostAvailPhysical != OMRPORT_MEMINFO_NOT_AVAILABLE)
       {
-      static int64_t lastUpdateTime = 0;
-      int64_t crtElapsedTime = getPersistentInfo()->getElapsedTime();
+      freePhysicalMemory = memInfo.availPhysical;
+      uint64_t freeHostPhysicalMemorySizeB = memInfo.hostAvailPhysical;
 
-      // time to recompute freePhysicalMemory
-      if (lastUpdateTime == 0
-          || (crtElapsedTime - lastUpdateTime) > (int64_t)TR::Options::getUpdateFreeMemoryMinPeriod())
-         {
-         PORT_ACCESS_FROM_JITCONFIG(_jitConfig);
+      if (memInfo.cached != OMRPORT_MEMINFO_NOT_AVAILABLE)
+         freePhysicalMemory += memInfo.cached;
+      else
+         incomplete = !_cgroupMemorySubsystemEnabled;
 
-         J9MemoryInfo memInfo;
-         if (0 == j9sysinfo_get_memory_info(&memInfo)
-             && memInfo.availPhysical != OMRPORT_MEMINFO_NOT_AVAILABLE)
-            {
-            uint64_t freePhysicalMemorySizeB = memInfo.availPhysical;
-           
-            if (memInfo.cached != OMRPORT_MEMINFO_NOT_AVAILABLE)
-               freePhysicalMemorySizeB += memInfo.cached;
-            else
-               incomplete = true;
+      if (memInfo.hostCached != OMRPORT_MEMINFO_NOT_AVAILABLE)
+         freeHostPhysicalMemorySizeB += memInfo.hostCached;
+      else
+         incomplete = true;
 #if defined(LINUX)
-            if (memInfo.buffered != OMRPORT_MEMINFO_NOT_AVAILABLE)
-               freePhysicalMemorySizeB += memInfo.buffered;
-            else
-               incomplete = true;
+      if (memInfo.buffered != OMRPORT_MEMINFO_NOT_AVAILABLE)
+         freePhysicalMemory += memInfo.buffered;
+      else
+         incomplete = !_cgroupMemorySubsystemEnabled;
```

Comment:
Is it possible for `incomplete` to be true from the line 10510, but then set to false in this line? If so, Shouldn't there be a check to not set it to false?

Reply:
You are right. I need to correct the behavior to never change it to 'false' once we set it to 'true'